app/main.py

```diff
@@ -1,2 +1,146 @@
+            if target is None:
+                raise KeyError(f"Key {key} does not exist")
+            if target != "del" and target[1] == key_hash and target[0] == key:
+                return target[2]
+            index = (index + 1) % self.capacity
+            if start_index == index:
+                raise KeyError(f"Key {key} does not exist")
+
+    def clear(self) -> None:
+        self.hash_table = [None] * self.capacity
+        self.length = 0
+
+    def __delitem__(self, key: Any) -> None:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        start_index = index
+
+        while True:
+            target = self.hash_table[index]
+            if target is None:
+                raise KeyError(f"Key {key} does not exist")
+            if target != "del" and target[1] == key_hash and target[0] == key:
+                self.hash_table[index] = "del"
+                self.length -= 1
+                break
+            index = (index + 1) % self.capacity
+            if start_index == index:
+                raise KeyError(f"Key {key} does not exist")
+
+    def get(self, key: Any, default: Any = None) -> Any:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        start_index = index
+
+        while True:
+            target = self.hash_table[index]
+            if target is None:
+                return default
+            if target != "del" and target[1] == key_hash and target[0] == key:
+                return target[2]
+            index = (index + 1) % self.capacity
+            if start_index == index:
+                return default
+
+    def pop(self, key: Any, default: Any = "raise") -> None:
+        key_hash = hash(key)
+        index = key_hash % self.capacity
+        start_index = index
+
+        while True:
+            target = self.hash_table[index]
+            if target is None:
+                if default == "raise":
+                    raise KeyError(f"Key {key} does not exist")
+                return default
+            if target != "del" and target[1] == key_hash and target[0] == key:
+                value = target[2]
+                self.hash_table[index] = "del"
+                self.length -= 1
+                return value
+            index = (index + 1) % self.capacity
+            if start_index == index:
+                if default == "raise":
+                    raise KeyError(f"Key {key} does not exist")
+                return default
+
+    def update(self, new_pairs: Iterable[Tuple] | Dict) -> None:
+        if isinstance(new_pairs, dict):
+            new_pairs = list(new_pairs.items())
+        for pair in new_pairs:
+            key, value = pair
+            self.__setitem__(key, value)
+
+    def __iter__(self) -> List:
+        keys_list = []
+        for bucket in self.hash_table:
+            if bucket is not None and bucket != "del":
+                keys_list.append(bucket[0])
+        return keys_list
```

Comment:
The `__iter__` method should return an iterator, not a list. You can achieve this by using `yield` to create a generator or by returning `iter(keys_list)` instead of `keys_list`.